src/services/member/plugins/profile/repository.ts

```diff
@@ -39,17 +50,35 @@ const MemberProfileRepository = AppDataSource.getRepository(MemberProfile).exten
     if (!memberId) {
       throw new MemberNotFound();
     }
-    const memberProfile = await this.findOne({
+    const memberProfile = await this.repository.findOne({
       where: { member: { id: memberId }, visibility: filter?.visibility },
       relations: ['member'],
     });
 
     return memberProfile;
-  },
+  }
+
+  async patch(memberId: string, data: Partial<IMemberProfile>): Promise<MemberProfile | null> {
+    /**
+     * this function is a test to try to make the queries more efficient while still
+     * getting the return values in one go.
+     * but this is really type unsafe, since we have to use the raw sql output provided by typeorm
+     * the normal `update` method from typeorm results in 3 Sql queries (because it needs to fetch the relation beforehand) and does not even allow us to get the resulting data.
+     *
+     * Not using the ORM means that the datetime columns are not updated from snake_case to camelCase, so they are not present at validation
+     */
+    const {
+      raw: [profile],
+    } = await this.repository
+      .createQueryBuilder()
+      .update(MemberProfile)
+      .set(data)
+      .where('member.id = :memberId', { memberId })
+      .returning('*')
+      .execute();
 
-  async patch(memberId: string, data: Partial<IMemberProfile>): Promise<void> {
-    await this.update({ member: { id: memberId } }, data);
-  },
-});
+    return profile;
+  }
+}
```

Comment:
Do we really want to push that? If you want to keep it somewhere, it can be added to an issue or technical debt board.